apps/communications/contacts/views/matching/js/matching_ui.js

```diff
@@ -0,0 +1,492 @@
+'use strict';
+/* global ContactPhotoHelper */
+/* global ImageLoader */
+/* global LazyLoader */
+/* global Sanitizer */
+/* jshint nonew: false */
+
+(function(exports) {
+
+    var CONTACTS_APP_ORIGIN = location.origin;
+
+    var listDependencies = [
+      '/shared/js/contacts/utilities/image_loader.js'
+    ];
+
+    var MatchingUI = {
+      _: null,
+
+      // Counter for checked list items
+      checked: 0,
+
+      // Hash contains identifiers of checked contacts
+      checkedContacts: {},
+
+      // Field order when showing matching details main reason
+      fieldOrder: ['tel', 'email', 'name'],
+
+      init: function() {
+        this._ = navigator.mozL10n.get;
+        this.mergeButton = document.getElementById('merge-action');
+        if (!this.mergeButton) {
+          return;
+        }
+
+        this.duplicateMessage = document.querySelector('#duplicate-msg > p');
+        this.contactsList =
+          document.querySelector('#contacts-list-container > ol');
+        this.title = document.getElementById('title');
+
+        this.mergeHeader = document.getElementById('merge-header');
+        this.mergeHeader.addEventListener('action', this.onClose.bind(this));
+        this.contactsList.addEventListener('click', this.onClick.bind(this));
+        this.mergeButton.addEventListener('click', this.onMerge.bind(this));
+
+        this.matchingDetails = document.querySelector('#matching-details');
+        this.matchingFigure = document.querySelector('#matching-figure');
+        this.matchingDetailList =
+          this.matchingDetails.querySelector('#matching-list');
+        this.matchingImg = this.matchingDetails.querySelector('img');
+        this.matchingName = this.matchingDetails.querySelector('figcaption');
+
+        this.matchingDetails.querySelector('button').onclick = () => {
+          this.hideMatchingDetails();
+        };
```

Comment:
Why do you need to wrap the function instead of:
`this.matchingDetails.querySelector('button').onclick = this.hideMatchingDetails();`?